### PR TITLE
Upgrade from Chromium 125 to Chromium 126

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/37788

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by zenparsing — APPROVED** [review]
Rewards 👍

**Review by fmarier — APPROVED** [review]

**Review by tmancey — APPROVED** [review]
LGTM

File: patches/chrome-browser-ui-views-location_bar-location_bar_view.cc.patch
Comment (simonhong) [inline patches/chrome-browser-ui-views-location_bar-location_bar_view.cc.patch:9]: Just moved position to apply `icon_indent, text_indent` to our value.

File: patches/ui-views-layout-layout_provider.h.patch
Comment (simonhong) [inline patches/ui-views-layout-layout_provider.h.patch:10]: Need to override `GetCornerRAdiusMetric` from our subclass but upstream has another same name method with different params. and it's already `virtual`. So `#define GetCornerRadiusMetric virtual GetCornerRAdiusMetric` doesn't work.

File: patches/ui-views-controls-button-md_text_button.h.patch
Comment (simonhong) [inline patches/ui-views-controls-button-md_text_button.h.patch:10]: MdTextButton has its Builder `BEGIN_VIEW_BUILDER(VIEWS_EXPORT, MdTextButton, LabelButton)`.
and that builder has `SetStyle` method `VIEW_BUILDER_PROPERTY(ui::ButtonStyle, Style)`
`#define SetStyle virtual SetStyle` affects this builder's interface also. so directly patched.

**Review by AlexeyBarabash — COMMENTED** [review]
sync changes looks ok, 
basic sync functions work on desktop <=> Android for bookmarks and history.

however, I got failed DCHECK on Android while tried to enable all types
```
[FATAL:sync_user_settings_impl.cc(183)] Check failed: registered_types.HasAll(types). 
 registered: bookmarks, preferences, autofill, typedUrls, readingList, tabs, payments
 setting to: bookmarks, preferences, passwords, autofill, typedUrls, tabs, payments 
```

I'll approve when I'll fix it.

File: patches/chrome-browser-resources-side_panel-bookmarks-power_bookmark_row.html.patch
Comment (mkarolin) [inline patches/chrome-browser-resources-side_panel-bookmarks-power_bookmark_row.html.patch:1]: Was this patch meant to just be deleted?

File: patches/chrome-browser-resources-side_panel-bookmarks-power_bookmark_row.ts.patch
Comment (mkarolin) [inline patches/chrome-browser-resources-side_panel-bookmarks-power_bookmark_row.ts.patch:1]: Was this patch meant to just be deleted?

File: patches/chrome-browser-shell_integration.h.patch
Comment (mkarolin) [inline patches/chrome-browser-shell_integration.h.patch:9]: Shouldn't delete upstream empty lines

File: patches/chrome-browser-resources-side_panel-bookmarks-power_bookmark_row.html.patch
Comment (mkarolin) [reply patches/chrome-browser-resources-side_panel-bookmarks-power_bookmark_row.html.patch:1]: I see, this was an upstream patch that got deleted.

File: patches/chrome-browser-resources-side_panel-bookmarks-power_bookmark_row.ts.patch
Comment (mkarolin) [reply patches/chrome-browser-resources-side_panel-bookmarks-power_bookmark_row.ts.patch:1]: I see, this was an upstream patch that got deleted.

**Comment by AlexeyBarabash** [comment]
> however, I got failed DCHECK on Android while tried to enable all types
> 

DCHECK is fixed with https://github.com/brave/brave-core/pull/23233/commits/49ff06ecba3ef63a7adaeb45a0205c67ed0161e2 , also this fixed the password manager - now it prompts to save passwords, before it didn't.

Now I see another issue: sync of passwords works bidirectional for Desktop <=> Desktop and only one way for Desktop => Android. 

Android => Android or Android => Desktop doesn't work. figuring this out.

File: ios/app/brave_core_main.mm
Comment (mkarolin) [inline ios/app/brave_core_main.mm:218]: Do we need to create a follow up issue to rework these to whatever the new method is?

Comment (emerick) [reply ios/app/brave_core_main.mm:218]: Created https://github.com/brave/brave-browser/issues/38419 to track this. cc: @kylehickinson

File: chromium_src/ui/views/controls/button/md_text_button.cc
Comment (mkarolin) [inline chromium_src/ui/views/controls/button/md_text_button.cc]: Need a follow up issue?

File: patches/chrome-browser-shell_integration.h.patch
Comment (emerick) [reply patches/chrome-browser-shell_integration.h.patch]: Thanks, fixed in 2a3ba3e1d379335edcff06fbff44b4f7bc9e5bcf.

File: chromium_src/ui/views/controls/button/md_text_button.cc
Comment (emerick) [reply chromium_src/ui/views/controls/button/md_text_button.cc]: cc: @simonhong?

File: chromium_src/google_apis/google_api_keys-inc.cc
Comment (mkarolin) [inline chromium_src/google_apis/google_api_keys-inc.cc:6]: Do we not need this include any more? If so, do we even need this override?

Comment (emerick) [reply chromium_src/google_apis/google_api_keys-inc.cc:6]: cc: @cdesouza-chromium

File: app/feature_defaults_unittest.cc
Comment (mkarolin) [inline app/feature_defaults_unittest.cc:221]: Looks like this was meant to be in a separate commit (along with chromium_src/components/metrics/structured/structured_metrics_features.cc)?

**Review by mkarolin — APPROVED** [review]
`strings`++

File: app/feature_defaults_unittest.cc
Comment (emerick) [reply app/feature_defaults_unittest.cc:221]: Removed here 6692557058cf95785c387853e04daa396c6aad4f and added back in separate commit here d12e0bf2e8d430b0f584446bb10d02cf4433e196

**Comment by AlexeyBarabash** [comment]
> Android => Android or Android => Desktop doesn't work. figuring this out.

It was false alarm, the reason was in weak internet connection on my Android test device. Now I see no issues with sync, approving.

**Review by AlexeyBarabash — APPROVED** [review]
sync++

File: patches/chrome-browser-ui-startup-startup_browser_creator_impl.cc.patch
Comment (bridiver) [inline patches/chrome-browser-ui-startup-startup_browser_creator_impl.cc.patch:18]: is this a wip? These patches definitely do not follow patching standards

File: chromium_src/components/feed/core/shared_prefs/pref_names.cc
Comment (bridiver) [inline chromium_src/components/feed/core/shared_prefs/pref_names.cc:26]: don't we normally just call `SetDefaultPrefValue` in `brave::RegisterProfilePrefs` for this kind of thing?

**Review by darkdh — APPROVED** [review]

File: patches/third_party-lit-v3_0-BUILD.gn.patch
Comment (bridiver) [inline patches/third_party-lit-v3_0-BUILD.gn.patch:9]: this does not follow patching standards, you should have `visibility += brave_lit_build_ts_visibility`

File: patches/ui-views-layout-layout_provider.h.patch
Comment (bridiver) [reply patches/ui-views-layout-layout_provider.h.patch:10]: why do we need to override this one? Doesn't it defer to the other one for basically everything when `IsChromeRefresh2023` is not enabled?

File: chromium_src/components/feed/core/shared_prefs/pref_names.cc
Comment (emerick) [reply chromium_src/components/feed/core/shared_prefs/pref_names.cc:26]: cc: @AlexeyBarabash

File: patches/third_party-lit-v3_0-BUILD.gn.patch
Comment (emerick) [reply patches/third_party-lit-v3_0-BUILD.gn.patch:9]: cc: @fallaciousreasoning

File: patches/third_party-polymer-v3_0-BUILD.gn.patch
Comment (bridiver) [inline patches/third_party-polymer-v3_0-BUILD.gn.patch:17]: this patch does not follow patching standards. It should be `visibility += brave_polymer_library_visibility`

Comment (emerick) [reply patches/third_party-polymer-v3_0-BUILD.gn.patch:17]: cc: @fallaciousreasoning

File: build/commands/lib/config.js
Comment (bridiver) [inline build/commands/lib/config.js:599]: I'm confused, why do we not want the same behavior as upstream here?

Comment (emerick) [reply build/commands/lib/config.js:599]: cc: @AlexeyBarabash

File: patches/components-permissions-permission_uma_util.cc.patch
Comment (bridiver) [inline patches/components-permissions-permission_uma_util.cc.patch:13]: @goodov do we actually need this stuff for anything? Why don't we just override PermissionUmaUtil to be a no-op for everything?

File: patches/chrome-browser-resources-side_panel-read_anything-BUILD.gn.patch
Comment (bridiver) [inline patches/chrome-browser-resources-side_panel-read_anything-BUILD.gn.patch:10]: is there a follow-up to this?

File: patches/chrome-android-java-src-org-chromium-chrome-browser-homepage-HomepageManager.java.patch
Comment (bridiver) [inline patches/chrome-android-java-src-org-chromium-chrome-browser-homepage-HomepageManager.java.patch:9]: we can't make this public with asm?

File: chromium_src/ui/views/controls/button/md_text_button.cc
Comment (bridiver) [inline chromium_src/ui/views/controls/button/md_text_button.cc:304]: I don't understand what we're doing here, why do we have `Kind` that maps directly to upstream styles? Why wouldn't we just use the upstream style names?

File: build/commands/lib/config.js
Comment (AlexeyBarabash) [reply build/commands/lib/config.js:599]: because now without that flag Chrome uses OS-provided password store. If so, we lose ability to sync passwords with our Sync, because Chrome uses Google Services and Google account to sync it.

File: patches/ui-views-controls-button-md_text_button.h.patch
Comment (bridiver) [reply patches/ui-views-controls-button-md_text_button.h.patch:10]: I don't understand why we need to override `SetStyle` at all just to set some `Kind` thing that maps directly to the style. Why aren't we just using the style directly?

File: patches/chrome-android-java-src-org-chromium-chrome-browser-homepage-HomepageManager.java.patch
Comment (AlexeyBarabash) [reply patches/chrome-android-java-src-org-chromium-chrome-browser-homepage-HomepageManager.java.patch:9]: we can, but we need to have it public during compile time to subclass with [BraveHomepageManager](https://github.com/brave/brave-core/pull/23233/files?file-filters%5B%5D=.chromium&file-filters%5B%5D=.flags&file-filters%5B%5D=.gn&file-filters%5B%5D=.gni&file-filters%5B%5D=.grd&file-filters%5B%5D=.icon&file-filters%5B%5D=.java&file-filters%5B%5D=.js&file-filters%5B%5D=.json5&file-filters%5B%5D=.lock&file-filters%5B%5D=.mm&file-filters%5B%5D=.py&file-filters%5B%5D=.toml&file-filters%5B%5D=.ts&file-filters%5B%5D=No+extension&show-deleted-files=true&show-viewed-files=true#diff-be074d694bc3f105d40784fc92a3157bae4200b532472c56282285f9b7a54266R11)

File: patches/chrome-browser-resources-side_panel-read_anything-BUILD.gn.patch
Comment (emerick) [reply patches/chrome-browser-resources-side_panel-read_anything-BUILD.gn.patch:10]: Filed https://github.com/brave/brave-browser/issues/38422.

File: chromium_src/ui/views/controls/button/md_text_button.cc
Comment (emerick) [reply chromium_src/ui/views/controls/button/md_text_button.cc:304]: cc: @simonhong

File: build/commands/lib/util.js
Comment (bridiver) [inline build/commands/lib/util.js:69]: can we write this somewhere else, call it in run hooks and override `clang_unsafe_buffers_paths`?

Comment (emerick) [reply build/commands/lib/util.js:69]: cc: @cdesouza-chromium

Comment (cdesouza-chromium) [reply build/commands/lib/util.js:69]: After discussions with @bridiver I'm considering having this running as a hook. @goodov Any thoughts?

File: chromium_src/components/feed/core/shared_prefs/pref_names.cc
Comment (AlexeyBarabash) [reply chromium_src/components/feed/core/shared_prefs/pref_names.cc]: thanks @bridiver , your option is indeed better,

fixed at https://github.com/brave/brave-core/pull/23233/commits/e99d97f19f23286d05202facb04650e66fcadb5a

File: patches/chrome-browser-ui-startup-startup_browser_creator_impl.cc.patch
Comment (emerick) [reply patches/chrome-browser-ui-startup-startup_browser_creator_impl.cc.patch]: Fixed here 51f8840abedea20b807f9d16864f8a15af410c82 using your advice to override `GetDistributionFirstRunTabs`.

File: patches/third_party-lit-v3_0-BUILD.gn.patch
Comment (fallaciousreasoning) [reply patches/third_party-lit-v3_0-BUILD.gn.patch:9]: Fixed!

File: patches/third_party-polymer-v3_0-BUILD.gn.patch
Comment (fallaciousreasoning) [reply patches/third_party-polymer-v3_0-BUILD.gn.patch]: Fixed!

File: chromium_src/ui/views/controls/button/md_text_button.cc
Comment (simonhong) [reply chromium_src/ui/views/controls/button/md_text_button.cc:304]: Right, we could `ui::ButtonStyle` instead of `Kinde` now. then Patch file for `SetStyle` doesn't need anymore. Working on it.

Comment (simonhong) [reply chromium_src/ui/views/controls/button/md_text_button.cc]: Removed `Kind` via  a951820f95fdade377b64fcc27947f494bb0682a

File: build/commands/lib/util.js
Comment (goodov) [reply build/commands/lib/util.js:69]: > override `clang_unsafe_buffers_paths`

That's fine, but not sure if this should be part of hooks. I'd rather keep this code in build scripts so the path to a generated file is shared between the code that generates it and the arg we pass to GN.

File: build/commands/lib/updateUnsafeBuffersPaths.js
Comment (emerick) [reply build/commands/lib/updateUnsafeBuffersPaths.js:29]: @cdesouza-chromium Any concerns here?

Comment (cdesouza-chromium) [reply build/commands/lib/updateUnsafeBuffersPaths.js:29]: Nope.

File: ios/browser/favicon/brave_ios_favicon_loader.mm
Comment (StephenHeaps) [inline ios/browser/favicon/brave_ios_favicon_loader.mm:1]: @brave/ios `DCHECK` hit on fresh launch in `BraveFaviconLoader`
https://github.com/brave/brave-core/blob/cee7c87be815f3dcee3a5c80336c748a4de58a15/ios/browser/favicon/brave_ios_favicon_loader.mm#L99

**Review by samartnik — APPROVED** [review]
java lgtm

File: ios/browser/favicon/brave_ios_favicon_loader.mm
Comment (emerick) [reply ios/browser/favicon/brave_ios_favicon_loader.mm:1]: Maybe triggered by this change 276bc7615efed7af2b066d5fb89ddda45ce7f154? Could definitely use some advice on what needs to be done here.

File: patches/third_party-lit-v3_0-BUILD.gn.patch
Comment (emerick) [reply patches/third_party-lit-v3_0-BUILD.gn.patch:9]: @fallaciousreasoning I'm not sure what's going on, but it seems that `npm run sync` always reapplies this particular patch.

File: ios/browser/favicon/brave_ios_favicon_loader.mm
Comment (kylehickinson) [inline ios/browser/favicon/brave_ios_favicon_loader.mm:56]: This should be `kReturnFallbackColor` to match Chrome iOS

**Review by kylehickinson — CHANGES_REQUESTED** [review]

File: patches/third_party-lit-v3_0-BUILD.gn.patch
Comment (emerick) [reply patches/third_party-lit-v3_0-BUILD.gn.patch:9]: @cdesouza-chromium sorted this out.

File: ios/browser/favicon/brave_ios_favicon_loader.mm
Comment (emerick) [reply ios/browser/favicon/brave_ios_favicon_loader.mm]: Fixed in 9451a06b39880b9709b57a6e76f75ad0b0fc369d. That should fix the `DCHECK` we were seeing here?

**Review by kylehickinson — APPROVED** [review]
iOS++

**Review by StephenHeaps — APPROVED** [review]
iOS wallet ++

File: chromium_src/google_apis/google_api_keys-inc.cc
Comment (mkarolin) [reply chromium_src/google_apis/google_api_keys-inc.cc:6]: @cdesouza-chromium, sorry, somehow I missed the non-changed part of the diff, so I thought the only change was to rename and remove the header. Please, ignore my original questions.

File: patches/chrome-browser-resources-side_panel-read_anything-BUILD.gn.patch
Comment (bridiver) [reply patches/chrome-browser-resources-side_panel-read_anything-BUILD.gn.patch:10]: Well, I meant a follow-up for the chromium update ideally. Not optimizing this doesn't seem great cc @petemill

File: build/commands/lib/util.js
Comment (bridiver) [reply build/commands/lib/util.js:69]: The value can be shared through gclient_args.gni. We used to do the same thing with the rust version in our old rust build process

**Review by rillian — APPROVED** [review]
rust++

File: patches/chrome-browser-resources-side_panel-read_anything-BUILD.gn.patch
Comment (petemill) [reply patches/chrome-browser-resources-side_panel-read_anything-BUILD.gn.patch:10]: We don't even use this WebUI, so not sure we need to invest time finding a fix for the cause yet?

File: patches/ui-views-layout-layout_provider.h.patch
Comment (bridiver) [reply patches/ui-views-layout-layout_provider.h.patch:10]: https://google.github.io/styleguide/cppguide.html#Default_Arguments `Default arguments are banned on virtual functions, where they don't work properly`

we could just patch the cc file and do the same thing without the above issue so you might as well just patch directly. This is going to change soon anyway with `IsChromeRefresh2023`

Comment (simonhong) [reply patches/ui-views-layout-layout_provider.h.patch]: Addressed - c82707a4c30d30d8b23c3c8420ce471e86e89d97

File: build/commands/lib/util.js
Comment (goodov) [reply build/commands/lib/util.js:69]: > The value can be shared through gclient_args.gni. We used to do the same thing with the rust version in our old rust build process

`gclient_args.gni` (or `brave_gclient_args.gni`) should be imported at the right place to override `clang_unsafe_buffers_paths`. We either need to patch `BUILDCONFIG.gn` or import the generated file manually via `args_generated.gni`.

I agree we should make use of `clang_unsafe_buffers_paths` in this PR, but a possible move to hooks looks like a follow up to me.

**Review by goodov — APPROVED** [review]

**Review by mherrmann — APPROVED** [review]
The Keystone changes look fine to me.

File: patches/chrome-browser-shell_integration.h.patch
Comment (emerick) [reply patches/chrome-browser-shell_integration.h.patch:9]: I'm fine with moving it, but the current advice on our wiki is:

> The define should always be the last thing in public so you can change to protected or private inside the define.

https://github.com/brave/brave-browser/wiki/Patching-Chromium

Comment (goodov) [reply patches/chrome-browser-shell_integration.h.patch:9]: yeah, this comment was more about the rule we have, not about this exact patch.

Maybe we should update the approach in such cases a bit.

Comment (emerick) [reply patches/chrome-browser-shell_integration.h.patch:9]: Gotcha, I'll create a thread on Slack to discuss updating this approach.

**Review by bridiver — APPROVED** [review]

**Review by iefremov — APPROVED** [review]

File: build/commands/lib/util.js
Comment (cdesouza-chromium) [reply build/commands/lib/util.js:69]: Fixed in 72935603d2077749971b9da2a8efa8bdb274c064. This change makes use of the gn arg `clang_unsafe_buffers_paths`. An issue has been created as a follow up to move this code to `runhooks` https://github.com/brave/brave-browser/issues/38512

**Pull request merged**
